src/core.js

```diff
@@ -97,18 +98,38 @@ nv.render = function render(step) {
 
         nv.render.queue.splice(0, i);
 
-        if (nv.render.queue.length) setTimeout(arguments.callee, 0);
+        if (nv.render.queue.length) {
+            setTimeout(renderLoop, 0);
+        }
         else {
             nv.dispatch.render_end();
             nv.render.active = false;
         }
-    }, 0);
+    };
+
+    setTimeout(renderLoop,0);
 };
 
 nv.render.active = false;
 nv.render.queue = [];
 
-// main function to use when adding a new graph, see examples
+/*
+Adds a chart to the async rendering queue. This method can take arguments in two forms:
+nv.addGraph({
+    generate: <Function>
+    callback: <Function>
+})
+
+or 
+
+nv.addGraph(<generate Function>, <callback Function>)
+
+The generate function should contain code that creates the NVD3 model, sets options
+on it, adds data to an SVG element, and invokes the chart model. The generate function
+should return the chart model.  See examples/lineChart.html for a usage example.
+
+The callback function is optional, and it is called when the generate function completes.
+*/
```

Comment:
This is great work, but let's try to keep completely separate pieces of work in separate commits.